File: output/gemma4-e4b/avatar/Java/Python/atcoder_ABC172_D.py
import sys

def solve():
    # Read n from standard input
    try:
        # Read all input and strip whitespace, then convert to int
        input_data = sys.stdin.read().strip()
        if not input_data:
            return
        n = int(input_data)
    except Exception:
        # Handle case where input might be missing or malformed
        return

    ans = 0
    
    # Loop i from 1 to n (inclusive)
    for i in range(1, n + 1):
        # Calculate k = floor(n / i) using integer division
        k = n // i
        
        # Calculate the term: i * k * (k + 1) / 2
        # We use integer division // throughout to maintain integer arithmetic
        term = i * k * (k + 1) // 2
        
        ans += term
        
    print(ans)

solve()

File: output/gemma4-e4b/avatar/Python/Java/atcoder_ABC172_D.java
import java.util.Scanner;

public class atcoder_ABC172_D {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        // Read N (equivalent to N = ri())
        if (!scanner.hasNextLong()) {
            return;
        }
        long N = scanner.nextLong();
        
        long ans = 0;
        
        // Loop from i = 1 to N
        for (long i = 1; i <= N; i++) {
            // Calculate k = N // i (integer division)
            long k = N / i;
            
            // Calculate the term: i * k * (k + 1) / 2
            // We calculate k * (k + 1) / 2 first to ensure intermediate results fit in long
            long sum_k = k * (k + 1) / 2;
            
            long term = i * sum_k;
            
            ans += term;
        }
        
        System.out.println(ans);
        
        scanner.close();
    }
}